### PR TITLE
Remove redundant None check from spawn plugins

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Part of #10034, #10059.

There is a None check in `DDPSpawnPlugin` and `TPUSpawnPlugin` that never evaluates to true, because the `self.mp_queue` is never None inside the spawned processes. 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

Part of #1 (it's a lie, this is just here to avoid noisy GitHub bot)

cc @borda @kaushikb11 @justusschock @awaelchli @akihironitta

### CODE REVIEW

File: pytorch_lightning/plugins/training_type/ddp_spawn.py [hunk 1]
```diff
@@ -271,15 +272,14 @@ def determine_ddp_device_ids(self):
         return [self.root_device.index]
 
     def __transfer_distrib_spawn_state_on_fit_end(self, trainer: "pl.Trainer", results: Any) -> None:
+        rank_zero_info("Transferring results back to main process ...")
```
Comment (carmocca) [inline pytorch_lightning/plugins/training_type/ddp_spawn.py:275]: Did you consider making this debug?

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/plugins/training_type/ddp_spawn.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/plugins/training_type/ddp_spawn.py:275]: I don't see a big difference. Could be any of the two. Definitely not a warning though.

Comment (carmocca) [reply pytorch_lightning/plugins/training_type/ddp_spawn.py:275]: Should all spawn users see this? IMO no

Comment (tchaton) [reply pytorch_lightning/plugins/training_type/ddp_spawn.py:275]: Let's make it debug. I got some complaints about it.

File: pytorch_lightning/plugins/training_type/tpu_spawn.py [hunk 2]
```diff
@@ -209,27 +209,25 @@ def barrier(self, name: Optional[str] = None) -> None:
...
-                # todo, pass complete checkpoint as state dictionary
-                self.mp_queue.put(best_model_path)
-                self.mp_queue.put(last_path)
-                self.mp_queue.put(results)
-                self.lightning_module.add_to_queue(self.mp_queue)  # adds the `callback_metrics` to the queue
+        # save the last weights
+        last_path = None
+        if trainer.state.fn == TrainerFn.FITTING and best_model_path is not None and len(best_model_path) > 0:
+            last_path = re.sub(".ckpt", ".tmp_end.ckpt", best_model_path)
+            self.checkpoint_io.save_checkpoint(state_dict, last_path)
```
Comment (tchaton) [inline pytorch_lightning/plugins/training_type/tpu_spawn.py:223]:
```suggestion
            tmp_ckpt_path = re.sub(".ckpt", ".tmp_end.ckpt", best_model_path)
            self.checkpoint_io.save_checkpoint(state_dict, tmp_ckpt_path)
```

File: pytorch_lightning/plugins/training_type/ddp_spawn.py [hunk 3]
```diff
@@ -271,15 +272,14 @@ def determine_ddp_device_ids(self):
...
         checkpoint_callback = trainer.checkpoint_callback
         best_model_path = checkpoint_callback.best_model_path if checkpoint_callback else None
 
         # requires to compute the state_dict on all processes in case Metrics are present
         state_dict = self.lightning_module.state_dict()
 
-        if self.global_rank == 0 and self.mp_queue is not None:
-            rank_zero_warn("cleaning up ddp environment...")
-
+        if self.global_rank == 0:
```
Comment (tchaton) [inline pytorch_lightning/plugins/training_type/ddp_spawn.py:282]:
```suggestion
        if not self.global_rank == 0:
            return
```

File: pytorch_lightning/plugins/training_type/tpu_spawn.py [hunk 4]
```diff
@@ -209,27 +209,25 @@ def barrier(self, name: Optional[str] = None) -> None:
...
+
+        if self.local_rank == 0:
+            # todo, pass complete checkpoint as state dictionary
+            self.mp_queue.put(best_model_path)
```
Comment (tchaton) [inline pytorch_lightning/plugins/training_type/tpu_spawn.py:227]: Do we want to pass this as 1 argument instead ?

Comment (awaelchli) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:227]: Yes this code is subject to change in the future PR #10034. 
`self.mp_queue` will be removed and all results will be passed as one tuple.

File: pytorch_lightning/plugins/training_type/ddp_spawn.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/plugins/training_type/ddp_spawn.py:275]: Fine! 

#10864

File: pytorch_lightning/plugins/training_type/tpu_spawn.py (hunk 2 above)
Comment (awaelchli) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:223]: I will change it in the follow up #10034. that will anyway touch this section

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/plugins/training_type/tpu_spawn.py [hunk 5]
```diff
@@ -209,27 +209,25 @@ def barrier(self, name: Optional[str] = None) -> None:
...
-                self.mp_queue.put(best_model_path)
-                self.mp_queue.put(last_path)
-                self.mp_queue.put(results)
-                self.lightning_module.add_to_queue(self.mp_queue)  # adds the `callback_metrics` to the queue
+        # save the last weights
+        last_path = None
+        if trainer.state.fn == TrainerFn.FITTING and best_model_path is not None and len(best_model_path) > 0:
+            last_path = re.sub(".ckpt", ".tmp_end.ckpt", best_model_path)
+            self.checkpoint_io.save_checkpoint(state_dict, last_path)
+
```
Comment (kaushikb11) [inline pytorch_lightning/plugins/training_type/tpu_spawn.py:224]:
```suggestion
# We use `local_rank` here as separate filesystems are used for each VM for TPU Pod Training
```

**Review by kaushikb11 — APPROVED** [review]
LGTM!

File: pytorch_lightning/plugins/training_type/tpu_spawn.py (hunk 5 above)
Comment (awaelchli) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:224]: yep, theoretically we should use the `should_rank_save_checkpoint` property here and in the other plugin.
